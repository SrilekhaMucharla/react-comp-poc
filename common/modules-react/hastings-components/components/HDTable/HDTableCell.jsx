/* eslint-disable react/no-array-index-key */
import PropTypes from 'prop-types';
import React from 'react';


const HDTableCell = ({
    label,
    value,
    topDescription,
    bottomDescription,
    extraLines,
    boldText,
    style
}) => (
    <div label={label} style={style} className="hd-table-cell">
        {label && (
            <div className="hd-table-cell__label">
                {label}
            </div>
        )}
        <div>
            {topDescription && (
                <div>
                    {topDescription}
                </div>
            )}
            {(typeof value === 'boolean') ? (
                <i className={`fas fa-${(value) ? 'check' : 'times'} fa-lg`} />
            ) : (
                <div className="hd-table-cell__value">
                    {(typeof value === 'number') ? value.toLocaleString('en', { minimumFractionDigits: 2, maximumFractionDigits: 2 }) : value}
                </div>
            )}
            {bottomDescription && (
                <div className="hd-table-cell__bottom-descr">
                    {bottomDescription}
                </div>
            )}
        </div>
        {!!extraLines.length && (
            <div className="hd-table-cell__extra-line">
                {extraLines.map((line, i) => <div key={i}>{line}</div>)}
            </div>
        )}
        <div className="hd-table-cell__bold-text">
            {boldText}
        </div>
    </div>
);

HDTableCell.propTypes = {
    label: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
    topDescription: PropTypes.string,
    bottomDescription: PropTypes.string,
    extraLines: PropTypes.arrayOf(PropTypes.string),
    boldText: PropTypes.string,
    style: PropTypes.shape({
        gridRow: PropTypes.string,
        gridColumn: PropTypes.string
    })
};

HDTableCell.defaultProps = {
    topDescription: null,
    bottomDescription: null,
    extraLines: [],
    boldText: null,
    label: null,
    style: undefined
};

export default HDTableCell;
