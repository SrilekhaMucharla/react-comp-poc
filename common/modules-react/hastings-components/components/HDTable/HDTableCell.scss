.hd-table-cell {
    align-self: center;
    min-height: 45px;
    padding-top: 10px;
    padding-bottom: 10px;
    text-align: center;
    font-size: 12px;
    font-weight: 400;
    line-height: 14px;
    color: $app-theme;

    .fa-check,
    .fa-times {
        color: $mono-white;
        width: 24px;
        height: 24px;
        border-radius: 50%;
        display: inline-flex;
        justify-content: center;
        align-items: center;
    }

    .fa-check {
        background-color: $midnight-royal-blue;
    }

    .fa-check::before {
        content: "\f00c";
        font-size: 12px;
        @include media-breakpoint-up(md) {
            font-size: 14px;
        }
    }

    .fa-times {
        background-color: $brand-hastings-red;
    }

    .fa-times::before {
        content: "\f00d";
        font-size: 13px;
        @include media-breakpoint-up(md) {
            font-size: 15px;
        }
    }
}

.hd-table-cell__value {
    font-size: 18px;
    font-weight: 700;
    line-height: 20px;
}

.hd-table-cell__bold-text {
    font-weight: 700;
    padding-bottom: 10px;
}

.hd-table-cell__label,
.hd-table-cell__bottom-descr,
.hd-table-cell__extra-line {
    padding-bottom: 10px;
}

.hd-table-cell__extra-line {
    font-size: 12px;
    @include media-breakpoint-up(md) {
        font-size: 16px;
    }
}
