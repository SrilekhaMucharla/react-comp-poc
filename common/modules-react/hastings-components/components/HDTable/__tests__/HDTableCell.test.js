import React from 'react';
import Enzyme, { shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDTableCell from '../HDTableCell';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDTableCell />', () => {
    it('render with mandatory props', () => {
        const wrapper = shallow(<HDTableCell value="Text value" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render with boolean value and all optional props', () => {
        const wrapper = shallow((
            <HDTableCell
                label="Label"
                value={false}
                topDescription="top"
                bottomDescription="bottom"
                extraLines={['line1', 'line2']}
                boldText="Text"
                selected={false} />
        ));
        expect(wrapper).toMatchSnapshot();
    });

    it('icon for value === true should be check', () => {
        const wrapper = render((
            <HDTableCell
                label="Label"
                value />
        ));
        const icon = wrapper.find('.fa-check');
        expect(icon).toBeDefined();
    });

    it('icon for value === false should be times', () => {
        const wrapper = render((
            <HDTableCell
                label="Label"
                value={false} />
        ));
        const icon = wrapper.find('.fa-times');
        expect(icon).toBeDefined();
    });
});
