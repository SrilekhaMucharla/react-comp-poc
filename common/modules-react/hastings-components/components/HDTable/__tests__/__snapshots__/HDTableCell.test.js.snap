// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDTableCell /> render with boolean value and all optional props 1`] = `
<div
  className="hd-table-cell"
  label="Label"
>
  <div
    className="hd-table-cell__label"
  >
    Label
  </div>
  <div>
    <div>
      top
    </div>
    <i
      className="fas fa-times fa-lg"
    />
    <div
      className="hd-table-cell__bottom-descr"
    >
      bottom
    </div>
  </div>
  <div
    className="hd-table-cell__extra-line"
  >
    <div
      key="0"
    >
      line1
    </div>
    <div
      key="1"
    >
      line2
    </div>
  </div>
  <div
    className="hd-table-cell__bold-text"
  >
    Text
  </div>
</div>
`;

exports[`<HDTableCell /> render with mandatory props 1`] = `
<div
  className="hd-table-cell"
  label={null}
>
  <div>
    <div
      className="hd-table-cell__value"
    >
      Text value
    </div>
  </div>
  <div
    className="hd-table-cell__bold-text"
  />
</div>
`;
